### DISCUSSION TITLE
Setting a default value for `off_screen`

### DISCUSSION THREAD

**Discussion by dstansby** [pyvista/pyvista#1427] [Q&A] [answered]
It's possible to use `plotter = pv.Plotter(off_screen=True)` to e.g. save a screenshot of a plotter without showing the actual plotting window. I was wondering if there was any way to set a global default value of this parameter, perhaps through a pyvista configuration file or an environment variable? If not, this is a feature request for that functionality 😃

Apologies if I missed something in the docs; if I did this is a request for improved documentation, which I would be happy to try myself.

Our use case is having a set of pyvista gallery examples that don't produce a plotting window when run on continuous integration, but do when a user runs the examples locally. It would be idea if we could set a configuration option so that `off_screen=True` for all our CI runs, but `off_screen=False` (the default) for users locally running the examples.

**Comment by akaszynski** [comment, marked as answer]
There's a few ways of doing this.

Within Python:
```py
import pyvista
pyvista.OFF_SCREEN
```

or, alternatively as an environment variable:
```
export PYVISTA_OFF_SCREEN=True
```

**Comment by dstansby** [comment]
👍 amazing, thanks a lot! I can take a look into where I expected this to be in the docs, and put in a PR to make it more obvious

**Comment by akaszynski** [comment]
> I can take a look into where I expected this to be in the docs, and put in a PR to make it more obvious

That would be really helpful.  Thanks!

**Comment by drew-parsons** [comment]
I didn't find the PYVISTA_OFF_SCREEN environment variable in the docs either.  It would help, for example, when running tests in batch mode, which otherwise hang waiting for an unseen plot window to be closed.  @dstansby please do make a PR for the docs.
